ui/component/channelThumbnail/view.jsx

```diff
@@ -6,18 +6,23 @@ import Gerbil from './gerbil.png';
 
 type Props = {
   thumbnail: ?string,
-  uri: string,
+  uri: ?string,
   className?: string,
   thumbnailPreview: ?string,
   obscure?: boolean,
 };
 
 function ChannelThumbnail(props: Props) {
   const { thumbnail, uri, className, thumbnailPreview, obscure } = props;
-
   // Generate a random color class based on the first letter of the channel name
-  const { channelName } = parseURI(uri);
-  const initializer = channelName.charCodeAt(0) - 65; // will be between 0 and 57
+  let initializer;
+  if (thumbnail) {
+    const { channelName } = parseURI(uri);
+    initializer = channelName.charCodeAt(0) - 65; // will be between 0 and 57
+  } else {
+    // if we want to default a thumbnail
+    initializer = Math.floor(Math.random() * 104729); // 10000th prime number
```

Comment:
Why did you change this?

Reply:
You need some way of generating the thumbnails on anonymous comments & claims.